www/js/GenWidget.js

```diff
@@ -0,0 +1,94 @@
+/* global Widget, NNE */
+/*
+  -----------
+     info
+  -----------
+
+  This Code Generator Widget extends the base Widget class to add methods
+  for more easily creating CSS generator Widgets.
+
+  -----------
+     usage
+  -----------
+
+  // this class inherits all the properties/methods of the base Widget class
+  // refer to www/js/Widget.js to see what those are
+  // or take a look at the wiki...
+  // https://github.com/netizenorg/netnet.studio/wiki/Creating-Widgets
+
+  this.codeField = this.createCodeField({
+    value: 'font-size 24px;',
+    change: (e) => this._codeFieldUpdate(e)
+  })
+
+  this.slider = w.createSlider({
+    background: '#f00',
+    min: 0,
+    max: 100,
+    label: 'PX'
+    change: (e) => this._fontSliderUpdate(e)
+  })
+
+*/
+class GenWidget extends Widget {
+  constructor (opts) {
+    super(opts)
+    console.log('genwidget constructed')
+  }
+
+  createCodeField (opts) {
+    const el = document.createElement('div')
+    const input = document.createElement('input')
+    input.setAttribute('type', 'text')
+    input.addEventListener('change', (e) => opts.change(e))
```

Comment:
an input's "change" events only fires when the text is "entered" (as in u submit it or press enter, rather than fireing every-time u type in a new key), what u want is the "input" event ([for example](https://netizenorg.github.io/netitor/#code/eJyFj8sKwjAQRfd+xbhKChJ/IO1GuxMUdOOyJGMbSJOahyDivzu21m1hYGZx5lyuXO+Pu8v1VEOXelutpHFDTrSjCmagA0B5FxPcDFoNJWivco8uiXvG8DyjRZV84Gz8YwXxNCMsGq3rB6EHExM6JEh1jWuRbYBjAWUFL2KnAG9RWN9y5h38Kfzq3gvOKXhBOUOzUW5//T4AkU/N))

that said, i've always found that sorta confusing, which is why i think it's still a good idea to use the word "change" in the API u create, meaning leave it like: 
```js
this.codeField = this.createCodeField({
  value: 'font-size: 24px;',
  change: (e) => this._codeFieldUpdate(e)
})
```

Reply:
cool that is good to know. both can be useful huh.. 
anyways now that that's working, this.parseCSS is not a function at FontSizeWidget.js:19:22.. i could understand if it returned undefined or null or a bad value, but why wouldn't it be a function?

Reply:
spelling (gets me every time) parseCSS? or parceCSS?